test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_service_handlers_test.clj

```diff
@@ -171,6 +173,41 @@
           (is (= (:status response) 200))
           (is (= (:body response) init-param-two)))))))
 
+(deftest websocket-test
+  (testing "Websocket handlers"
+    (with-app-with-config app
+      [jetty9-service]
+      jetty-plaintext-config
+      (let [s                     (get-service app :WebserverService)
+            add-websocket-handler (partial add-websocket-handler s)
+            path                  "/test"
+            connected             (atom 0)
+            server-messages       (atom [])
+            client-messages       (atom [])
+            handlers              {:on-connect (fn [ws]
+                                                 (ws-session/send! ws "Hello client!")
+                                                 (swap! connected inc))
+                                   :on-text    (fn [ws text]
+                                                 (ws-session/send! ws (str "You said: " text))
+                                                 (swap! server-messages conj text))}]
+        (add-websocket-handler handlers path)
+        (let [socket (ws-client/connect (str "ws://localhost:8080" path)
+                                        :on-receive (fn [text] (swap! client-messages conj text)))]
+          (ws-client/send-msg socket "Hello websocket handler")
+          (ws-client/send-msg socket "You look dandy")
+          ;; Websockets by their nature are a chat protocol so we have
+          ;; to wait a small amount of time for the responses sent by
+          ;; the :on-text handler to make it back before we can
+          ;; disconnect
+          (Thread/sleep 200)
```

Comment:
It would if we simplify this down to *one* message pair, but as it's two messages we need two futures?  An map of futures?

If just sending one pair of messages works then swapping this for a future is fine.